discovery_engine_core/core/src/storage/sqlite.rs

```diff
@@ -855,6 +860,14 @@ impl FeedbackScope for SqliteStorage {
 
         Ok(view)
     }
+
+    async fn update_source_reaction(&self, source: &str, liked: bool) -> Result<(), Error> {
+        match self.fetch_source_reaction(source).await? {
+            None => self.store_source_reaction(source, liked).await,
+            Some(reaction) if reaction == liked => self.update_source_weight(source).await,
+            _ => self.delete_source_reaction(source).await,
```

Comment:
❔ why are we deleting the source info if the `reaction` is different then `liked` param? how can we then change it from like to dislike?

Reply:
right, it's not so intuitive but that was the heuristic originally implemented in the python implementation, see [this comment](https://github.com/xaynetwork/xayn_discovery_engine/pull/441#discussion_r895509780). to change to a dislike, the user would then have to dislike the source again.

Reply:
ah, you're right on both :) it's a bit counterintuitive and also how it's implemented currently